controllers/storage/provision.go

```diff
@@ -191,13 +169,7 @@ func (m *MinIOReconciler) generateAzureSecret(labels map[string]string) *corev1.
 }
 
 func (m *MinIOReconciler) generateGcsSecret(labels map[string]string) *corev1.Secret {
-	data := map[string]string{
-		"bucket":        m.HarborCluster.Spec.Storage.Gcs.Bucket,
-		"encodedkey":    m.HarborCluster.Spec.Storage.Gcs.EncodedKey,
-		"rootdirectory": m.HarborCluster.Spec.Storage.Gcs.RootDirectory,
-		"chunksize":     m.HarborCluster.Spec.Storage.Gcs.ChunkSize,
-	}
-	dataJson, _ := json.Marshal(&data)
+	dataJson, _ := json.Marshal(m.HarborCluster.Spec.Storage.Gcs)
```

Comment:
Is it ok to ignore the error here? Can we make sure the `Spec.Storage` is valid? By validation webhook?

@soulseen